ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/Metric.scala

```diff
@@ -141,13 +193,43 @@ object Metric {
   }
 
   object DelayMetric {
+
+    def empty[T](
+        recordTimeFunction: T => Seq[Timestamp],
+        objectives: List[DelayObjective],
+        clock: Clock,
+    ): DelayMetric[T] =
+      DelayMetric(
+        recordTimeFunction = recordTimeFunction,
+        clock = clock,
+        objectives = objectives.map(objective => objective -> None).toMap,
+      )
+
     final case class Value(meanDelaySeconds: Option[Long]) extends MetricValue {
       override def formatted: List[String] =
         List(s"mean delay: ${meanDelaySeconds.getOrElse("-")} [s]")
     }
 
-    def empty[T](recordTimeFunction: T => Seq[Timestamp], clock: Clock): DelayMetric[T] =
-      DelayMetric(recordTimeFunction, clock)
+    sealed trait DelayObjective extends ServiceLevelObjective[Value]
+    object DelayObjective {
+      final case class MaxDelay(maxDelaySeconds: Long) extends DelayObjective {
+        override def isViolatedBy(metricValue: Value): Boolean =
+          metricValue.meanDelaySeconds.exists(_ > maxDelaySeconds)
+
+        override def moreViolatingOf(first: Value, second: Value): Value =
```

Comment:
This method is a member of `MaxDelay`, but it doesn't really have anything to do with it, does it? (e.g. it doesn't take into account the object's context  - `maxDelaySeconds`

Reply:
I've explained my motivation in the comment above.
I'll move out the logic out of the objective 👍